src/test/java/io/stargate/sgv2/jsonapi/api/v1/DeleteManyIntegrationTest.java

```diff
@@ -253,17 +257,69 @@ public void noFilter() {
           .body("errors", is(nullValue()));
     }
 
+    /**
+     * when there is an empty filter, it will delete all data in collection, like truncate
+     * operation. it will not be restricted by maxDeleteDocuments (20)
+     */
     @Test
-    public void noFilterMoreDataFlag() {
+    public void emptyFilter() {
       insert(25);
       String json =
           """
           {
             "deleteMany": {
+              "filter": {}
             }
           }
           """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("status.ok", is(1))
+          .body("status.deletedCount", is(nullValue()))
```

Comment:
Is there any way to also get the deleted count without having to execute a separate count operation? Our dccs say that `deleteMany()` returns a `deletedCount` that contains the number of docs deleted, it would be a bit clunky if `deletedCount` isn't set in some cases. I wouldn't say that missing `deletedCount` is a critical issue, more a nice-to-have to ensure the API is consistent. If we're unable to return `deletedCount` in the case of a truncate, we should document that.

Reply:
To accurately return the total number of rows, a separate count operation is necessary, which may reduce efficiency. Aaron suggested [here](https://github.com/stargate/jsonapi/issues/631) using -1 as the value for deletedCount. What are your thoughts on this approach?

Reply:
@amorton can you comment on this? I would like to get this merged if possible.